src/soar_sdk/views/template_filters.py

```diff
@@ -0,0 +1,223 @@
+            yield result
+            result = []
+    if result:
+        yield result
+
+
+def remove_empty(dictionary: dict) -> dict:
+    """Remove empty values from dictionary."""
+    return {k: v for k, v in dictionary.items() if v}
+
+
+def by_key(dictionary: dict, key: str) -> str:
+    """Get dictionary value by key."""
+    return dictionary.get(key, "")
+
+
+def by_nested_key(dictionary: dict, key: str) -> Optional[str]:
+    """Get nested dictionary value by space-separated key."""
+    split_key = key.split()
+    src = dictionary.get(split_key[0], None)
+    if src:
+        return src.get(split_key[1], "")
+    else:
+        return None
+
+
+def is_list(item: object) -> bool:
+    """Check if item is a list."""
+    return isinstance(item, list)
+
+
+def typeof(item: object) -> type:
+    """Get type of item."""
+    return type(item)
+
+
+def safe_intcomma(value: Union[str, int]) -> str:
+    """Format integer with commas, safely handling non-integers."""
+    try:
+        int(value)
+        if humanize:
+            return humanize.intcomma(value)
+        else:
+            return f"{int(value):,}"
+    except (ValueError, TypeError):
+        return str(value)
+
+
+def hash_function(value: str, salt: str = "") -> str:
+    """Hash a value with optional salt."""
+    salted = f"{salt}{value}"
+    return hashlib.sha512(salted.encode()).hexdigest()
+
+
+def startswith(s: str, s2: str) -> bool:
+    """Check if string starts with substring."""
+    return str(s).startswith(str(s2))
+
+
+def getattribute(obj: object, key: str) -> object:
+    """Get attribute from object."""
+    return getattr(obj, key, "")
+
+
+def superslug(s: str) -> str:
+    """Create a super slug from string."""
+    s = str(s).replace("-", "_").replace(" ", "_")
+    s = re.sub(r"\W", "", s)
+    return s
+
+
+def sformat(s: str, value: object) -> str:
+    """Format string with value."""
+    return str(s) % value
+
+
+def jslist(obj: list) -> str:
+    """Convert list to JavaScript array string."""
+    return "[" + ",".join([f"'{s}'" for s in obj]) + "]"
+
+
+def to_json(obj: object) -> str:
+    """Convert object to JSON string."""
+    return json.dumps(obj)
+
+
+def absval(obj: Union[int, float]) -> Union[int, float]:
+    """Get absolute value."""
+    return abs(obj)
+
+
+def commasplit(string: str, index: int) -> str:
+    """Split string by comma and get item at index."""
+    return str(string).split(",")[index]
+
+
+def slashsplit(string: str) -> list[str]:
+    """Split string by slash."""
+    return str(string).split("/")
+
+
+def strip_tenant_id(dictionary: dict, key: str) -> str:
+    """Strip tenant ID from dictionary value."""
+    x = dictionary.get(key, "")
+    if len(x):
+        return str(x).split("_")[0]
+    else:
+        return ""
+
+
+def bleach_clean(value: str, **kwargs: object) -> str:
+    """Clean HTML using bleach."""
+    if bleach:
+        return bleach.clean(value, **kwargs)
+    return value
```

Comment:
Again, I don't understand these `if {module}` checks that were in the platform code...
```suggestion
    return bleach.clean(value, **kwargs)
```

Reply:
Updated